ulib/experimental/Steel.ST.Effect.Ghost.fsti

```diff
@@ -0,0 +1,102 @@
+(*
+   Copyright 2020 Microsoft Research
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*)
+
+module Steel.ST.Effect.Ghost
+
+open Steel.Memory
+module T = FStar.Tactics
+include Steel.Effect.Common
+module STAG = Steel.ST.Effect.AtomicAndGhost
+module STA = Steel.ST.Effect.Atomic
+
+(*** STGhost effect ***)
+
+[@@ erasable; ite_soundness_by ite_attr]
+total
+reflectable
+effect {
+  STGhostBase (a:Type)
+             (framed:bool)
+             (opened_invariants:inames)
+             (o:observability)
+             (pre:pre_t)
+             (post:post_t a)
+             (req:Type0)
+             (ens:a -> Type0)
+  with { repr = STAG.repr;
+         return = STAG.return_;
+         bind = STAG.bind;
+         subcomp = STAG.subcomp;
+         if_then_else = STAG.if_then_else }
+}
+
+// [@@ erasable; ite_soundness_by ite_attr]
+// total
+// reflectable
+// new_effect STGhostBase = STAG.STAGCommon
```

Comment:
This was an interesting problem. If I define it this way, then when using .checked files for STAG, this file fails to typecheck because the definition seems to be triply universe polymorphic when it really isn't. I needed to redefine the effect as below to avoid the problem.

Will try to isolate and see if I can repro with a small example. headsup @aseemr